specs/electra/validator.md

```diff
@@ -189,18 +189,37 @@ def prepare_execution_payload(state: BeaconState,
 
 *[New in Electra]*
 
-1. The execution payload is obtained from the execution engine as defined above using `payload_id`. The response also includes a `execution_requests` entry containing a list of bytes. Each element on the list corresponds to one SSZ list of requests as defined
-in [EIP-7685](https://eips.ethereum.org/EIPS/eip-7685). The index of each element in the array determines the type of request.
+1. The execution payload is obtained from the execution engine as defined above using `payload_id`. The response also includes a `execution_requests` entry containing a list of bytes. Each element on the list corresponds to one SSZ list of requests as defined in [EIP-7685](https://eips.ethereum.org/EIPS/eip-7685). The first byte of each request is used to determine the request type. Requests must be ordered by request type in ascending order. As a result, there can only be at most one instance of each request type.
 2. Set `block.body.execution_requests = get_execution_requests(execution_requests)`, where:
 
 ```python
-def get_execution_requests(execution_requests: Sequence[bytes]) -> ExecutionRequests:
-    deposits = ssz_deserialize(List[DepositRequest, MAX_DEPOSIT_REQUESTS_PER_PAYLOAD], execution_requests[0])
-    withdrawals = ssz_deserialize(List[WithdrawalRequest, MAX_WITHDRAWAL_REQUESTS_PER_PAYLOAD], execution_requests[1])
-    consolidations = ssz_deserialize(List[ConsolidationRequest, MAX_CONSOLIDATION_REQUESTS_PER_PAYLOAD],
-                                     execution_requests[2])
-
-    return ExecutionRequests(deposits, withdrawals, consolidations)
+def get_execution_requests(execution_requests_list: Sequence[bytes]) -> ExecutionRequests:
+    requests = {
+        DEPOSIT_REQUEST_TYPE: {
+            "field": "deposits",
+            "type": List[DepositRequest, MAX_DEPOSIT_REQUESTS_PER_PAYLOAD],
+        },
+        WITHDRAWAL_REQUEST_TYPE: {
+            "field": "withdrawals",
+            "type": List[WithdrawalRequest, MAX_WITHDRAWAL_REQUESTS_PER_PAYLOAD],
+        },
+        CONSOLIDATION_REQUEST_TYPE: {
+            "field": "consolidations",
+            "type": List[ConsolidationRequest, MAX_CONSOLIDATION_REQUESTS_PER_PAYLOAD],
+        },
+    }
+
+    prev_request_type = None
+    execution_requests = ExecutionRequests()
+    for request in execution_requests_list:
+        request_type, request_data = request[0:1], request[1:]
+        assert request_type in requests, "unexpected request type"
+        assert len(request_data) != 0, "empty request data"
+        assert prev_request_type is None or prev_request_type < request_type, "not ascending order"
+        prev_request_type = request_type
+        deserialized_request = ssz_deserialize(requests[request_type]["type"], request_data)
+        setattr(execution_requests, requests[request_type]["field"], deserialized_request)
```

Comment:
I do think we should "unroll" this so we can skip `setattr`

something like

```python
if request_type == DEPOSIT_REQUEST_TYPE:
    execution_requests.deposits = ssz_deserialize(List[...], request_data)
elif request_type == WITHDRAWAL_REQUEST_TYPE:
    ...
else:
    ...
```

Reply:
then I think `requests` can just be `[DEPOSIT_REQUEST_TYPE, WITHDRAWAL_REQUEST_TYPE, CONSOLIDATION_REQUEST_TYPE]`